### DISCUSSION TITLE
apt-offline command is not working

### DISCUSSION THREAD

**Discussion by SyedaFarook** [rickysarraf/apt-offline#186] [Q&A]
Hello, apt-offline command is not working in Ubuntu 20.04 LTS.
1) "apt-offline set my.sig" gives the error
Gathering details needed for 'update' operation
Gathering details needed for 'upgrade' operation
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
ERROR: FATAL: Something is wrong with the APT system

2) When i download the updated packages manually and tried to install these packages folder using apt-offline it never updates,
The version has no changes even though it says synced.

root@inf-004:~# sudo apt-offline install packageupdate/ --verbose
Proceeding with installation
VERBOSE: {}
VERBOSE: Great!!! No bugs found for all the packages that were downloaded.

VERBOSE: APT Signature verification path is: ['/etc/apt/trusted.gpg.d/', '/etc/apt/trusted.gpg']
VERBOSE: Adding /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg to the apt-offline keyring
VERBOSE: Adding /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-archive.gpg to the apt-offline keyring
VERBOSE: Adding /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg to the apt-offline keyring
VERBOSE: Adding /etc/apt/trusted.gpg/ubuntu-keyring-2018-archive.gpg to the apt-offline keyring
VERBOSE: Adding /etc/apt/trusted.gpg/ubuntu-keyring-2012-archive.gpg to the apt-offline keyring
VERBOSE: Adding /etc/apt/trusted.gpg/ubuntu-keyring-2012-cdimage.gpg to the apt-offline keyring
tar_1.30+dfsg-7ubuntu0.20.04.2_amd64.deb file synced.
VERBOSE: /var/cache/apt/archives/tar_1.30+dfsg-7ubuntu0.20.04.2_amd64.deb file synced.
VERBOSE: tar_1.30+dfsg-7ubuntu0.20.04.2_amd64.deb file synced to APT.
git_1%3a2.25.1-1ubuntu3.2_amd64.deb file synced.
VERBOSE: /var/cache/apt/archives/git_1%3a2.25.1-1ubuntu3.2_amd64.deb file synced.
VERBOSE: git_1%3a2.25.1-1ubuntu3.2_amd64.deb file synced to APT.
ca-certificates_20210119~20.04.2_all.deb file synced.
VERBOSE: /var/cache/apt/archives/ca-certificates_20210119~20.04.2_all.deb file synced.
VERBOSE: ca-certificates_20210119~20.04.2_all.deb file synced to APT.
libpython3.8-minimal_3.8.10-0ubuntu1~20.04.2_amd64.deb file synced.

3)Even tried with the zip folder
root@inf-004:~# sudo apt-offline install packages.zip --verbose
VERBOSE: No changelog available
Proceeding with installation
VERBOSE: {}
VERBOSE: Great!!! No bugs found for all the packages that were downloaded.

Traceback (most recent call last):
  File "/usr/bin/apt-offline", line 28, in <module>
    main()
  File "/usr/lib/python3/dist-packages/apt_offline_core/AptOfflineCoreLib.py", line 2380, in main
    args.func(args)
  File "/usr/lib/python3/dist-packages/apt_offline_core/AptOfflineCoreLib.py", line 1968, in installer
    data = open(tempZipFile, 'wb')
IsADirectoryError: [Errno 21] Is a directory: '/tmp/tmpq6gdtxxg/packageupdate/'

Looking forward to the assisstance.

**Comment by rickysarraf** [comment]
Step 1 itself failed with an error internal to `apt`

```
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
ERROR: FATAL: Something is wrong with the APT system
```

You first need to investigate what is wrong with `apt` on your system.
